lib/YamahaReceiver/YamahaReceiverClient.js

```diff
@@ -149,7 +149,7 @@ class YamahaReceiverClient {
                             playInfo.playing = playbackInfoResult === 'Play';
                             playInfo.artist = Entities.decode(metaInfoResult['Artist'][0]);
                             playInfo.album = Entities.decode(metaInfoResult['Album'][0]);
-                            playInfo.track = Entities.decode(metaInfoResult['Track'][0]);
+                            playInfo.track = Entities.decode(metaInfoResult['Song'][0]);
```

Comment:
It's probably better to refactor fetching meta info to a separate function.

Something like:
```node
playInfo.track = this.getTrackFromMeta(metaInfoResult);
```

And then implement that function like this:
```node
getTrackFromMeta(metaInfoResult) {
    let track = null;

    if (typeof metaInfoResult['Track'] !== "undefined") {
        track = metaInfoResult['Track'][0];
    } else if (typeof metaInfoResult['Song'] !== "undefined") {
        track = metaInfoResult['Song'][0];
    }
    
    return Entities.decode(track);
}
```

Reply:
This way we also make it easier to fix additional edge-cases in the future.

Reply:
I'll tell you something fantastic: these are not two different versions of the API, it is a difference between the sources. "Song" is used by AirPlay, and "Track" is used by Spotify. I discovered this now because I tested the source input select from the cards yesterday, and now I suddenly got the opposite error. So the PlayInfo-parsing should be structured per input. The common parts should be general, and the differences should be connected to the source. This way the code is a lot more readable.

Reply:
Hmm, that will make the client quite a bit bigger. I'll sleep on it and maybe I'll have an idea of how to structure it tomorrow.

Reply:
Thanks for the debugging btw